BootloaderCorePkg/BootloaderCorePkg.dsc

```diff
@@ -252,6 +252,11 @@
 
   gPlatformCommonLibTokenSpaceGuid.PcdIppHashLibSupportedMask    | $(IPP_HASH_LIB_SUPPORTED_MASK)
 
+  !if $(SIGN_HASH_TYPE) == SHA2_256
+    gPlatformCommonLibTokenSpaceGuid.PcdCompSignHashAlg | 0x1
+  !elseif $(SIGN_HASH_TYPE) == SHA2_384
+    gPlatformCommonLibTokenSpaceGuid.PcdCompSignHashAlg | 0x2

```

Comment:
Why can't we let the BoardConfig.py to define the correct value and use it directly?   
  gPlatformCommonLibTokenSpaceGuid.PcdCompSignHashAlg | $(SIGN_HAHS_TYPE_VALUE)

Reply:
SIGN_HASH_TYPE is configured as string and its used in BuildLoader methods to pass hashAlg type.
It would be better to use same $SIGN_HASH_TYPE instead of defining a new parameter to BoardConfig,

Reply:
I think the other way.   We can define SIGN_HASH_TYPE as value.   In BoardConfig.py, when we need access to the string, we can use dict   SIGN_HASH[SIGN_HASH_TYPE].  In general, we don't want to use #if condition in DSC since we have directly control on the PCD values.

Reply:
agreed. I will modify to use dict.